### PR TITLE
Document subprotocol access token

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by vikram-redhat** [comment]
[Document: Authenticate websocket API requests using subprotocols, not query parameters](https://trello.com/c/821iB8pP/578-document-authenticate-websocket-api-requests-using-subprotocols-not-query-parameters)

File: architecture/additional_concepts/authentication.adoc [hunk 1]
```diff
@@ -75,7 +75,9 @@ methods:
 
 OAuth Access Tokens::
 - Obtained from the {product-title} OAuth server using the `_<master>_/oauth/authorize` and `_<master>_/oauth/token` endpoints.
-- Sent as an `Authorization: Bearer...` header or an `access_token=...` query parameter
+- Sent as an `Authorization: Bearer...` header
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authentication.adoc]: header.

File: architecture/additional_concepts/authentication.adoc [hunk 2]
```diff
@@ -75,7 +75,9 @@ methods:
...
+- Sent as an `access_token=...` query parameter for websocket requests prior to 3.6
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authentication.adoc]: ifdef::openshift-enterprise[]
prior to {product-title} 3.6.
endif::[]
ifdef::openshift-origin[]
prior to {product-title}1.6.
endif::[]

File: architecture/additional_concepts/authentication.adoc [hunk 3]
```diff
@@ -75,7 +75,9 @@ methods:
...
+- Sent as a websocket subprotocol header in the form `base64url.bearer.authorization.k8s.io.<base64url-encoded-token>` for websocket requests in versions 3.6 and after
```
Comment (ahardin-rh) [inline architecture/additional_concepts/authentication.adoc]: ifdef::openshift-enterprise[]
{product-title} versions 3.6 and later.
endif::[]
ifdef::openshift-origin[]
{product-title} versions 3.6 and later.
endif::[]

Comment (ahardin-rh) [inline architecture/additional_concepts/authentication.adoc]: I think may need to further conditionalize for Online and Dedicated. The added content may not be applicable to those products. Is that correct?

**Review by ahardin-rh — CHANGES_REQUESTED** [review]

File: architecture/additional_concepts/authentication.adoc (hunk 3 above)
Comment (liggitt) [inline architecture/additional_concepts/authentication.adoc]: it's just as applicable to those products as the access_token query param doc was

**Comment by ahardin-rh** [comment]
I am going to merge this to get this moving along, then do a follow-up PR for minor style edits. Thanks!

**Comment by ahardin-rh** [comment]
No revision history needed

**Pull request merged**